datafusion/core/src/physical_plan/joins/nested_loop_join.rs

```diff
@@ -375,13 +370,13 @@ struct NestedLoopJoinStream {
     /// Join execution metrics
     join_metrics: BuildProbeJoinMetrics,
     /// Memory reservation for visited_left_side
-    reservation: SharedMemoryReservation,
+    reservation: MemoryReservation,
 }
 
 fn build_join_indices(
     left_index: usize,
     batch: &RecordBatch,
```

Comment:
its out of the changes scope but probably we can rename too generic `batch` into something else? like right_data, or probe?

Reply:
https://github.com/apache/arrow-datafusion/pull/6229